### ISSUE TITLE
Country names are drawn twice

### ISSUE THREAD

**Issue by akovalov** [mousebird-consulting-inc/WhirlyGlobe#1516]
Some country names are drawn twice like on the screenshot. Looks like second label appears from some zoom level. 

<img src="https://user-images.githubusercontent.com/9693643/158218111-edbbd25c-210f-43e1-9ba1-4fc99d430d81.jpeg" width=300>

In generated .mbtiles file each country is assigned a "rank" number field and min zoom to show based on population like this

```
local place = node:Find("place")

		local rank = nil
		local mz = 13
		local pop = tonumber(node:Find("population")) or 0

		if     place == "continent"     then mz=0
		elseif place == "country"       then
			if     pop>50000000 then rank=1; mz=1
			elseif pop>20000000 then rank=2; mz=2
			else                     rank=3; mz=3 end
		else return
		end

		node:Layer("place", false)
		node:MinZoom(mz)
		if rank then node:AttributeNumeric("rank", rank) end
		if place=="city" then node:AttributeNumeric("population", pop) end
		SetNameAttributes(node)
```

This happens only for those country names where rank=1 and minZoom = 1. Other countries are OK. 

Style sheet is Mapbox-like and has just one layer for country labels like this

```
{
      "id": "country_label",
      "type": "symbol",
      "source": "openmaptiles",
      "source-layer": "place",
      "minzoom": 1,
      "maxzoom": 12,
      "filter": [
        "all",
        ["==", "$type", "Point"],
        ["==", "class", "country"]
      ],
      "layout": {
        "text-field": "{name:latin}",
        "text-font": ["Noto Sans Bold"],
        "text-max-width": 10,
        "text-size": {"stops": [[3, 12], [8, 22]]},
        "visibility": "visible",
        "text-allow-overlap": true
      },
      "paint": {
        "text-color": "hsl(0, 0%, 13%)",
        "text-halo-blur": 0,
        "text-halo-color": "rgba(255,255,255,0.75)",
        "text-halo-width": 2
      }
    }
```

Do you have any ideas why the second label can appear here?

**Comment by mousebird** [comment]
If you turn on text-allow-overlap you'll definitely get some overlap.  It's best to leave it on for vector styles unless you have a specific goal in mind that needs it off.

**Comment by akovalov** [comment]
Fixed it by setting `if     pop>50000000 then rank=1; mz=2`(mz=2 instead of mz=1) for largest population countries and this issue does not happen anymore. So this happened for some reason when minZoom was set to 1. But as far as I can't provide any more details if this is the issue inside lib or in mbtiles or my code, the question is closed. Thank you.
